File: 프로그래머스/2/42885. 구명보트/구명보트.py
from collections import deque

def solution(people, limit):
    
    result = 0
    queue = deque(sorted(people, reverse=True))
    
    while len(queue) > 1:
        if queue[0] + queue[-1] <= limit:
            result += 1
            queue.popleft()
            queue.pop()
        else:
            result += 1
            queue.popleft()
            
    if queue:
        result += 1
        
    return result

File: 프로그래머스/2/42885. 구명보트/구명보트.java
import java.util.*;

class Solution {
    public int solution(int[] people, int limit) {
        
        int answer = 0;
        
        // 정렬
        Arrays.sort(people);
        
        // 무거운 사람 태우고 > 가벼운 사람 태울 수 있으면 태우기
        int light = 0; int heavy = people.length - 1;
        while (light <= heavy) {
            if (people[light] <= limit - people[heavy]) {
                light++;
            }
            heavy--;
            answer++;
        }
        
        return answer;
    }
}